### PR TITLE
Require plugin IDs to be valid

### LINKED ISSUE #203: No Validation on the plugin creation script
At present, there is no validation on the plugin ID when creating a plugin on the command line.
I was able to create plugins with names:

- $ &&!@#%!#%SD_-?><~+
- ~

### PR DESCRIPTION
**Pull request by freben**
Closes #203

### CODE REVIEW

File: scripts/cli-e2e-test.js [hunk 1]
```diff
@@ -92,7 +112,7 @@ async function waitForPageWithText(
   browser,
   path,
   text,
-  { intervalMs = 1000, maxAttempts = 60 } = {},
+  { intervalMs = 1000, maxAttempts = 120 } = {},
```
Comment (Rugvip) [inline scripts/cli-e2e-test.js]: Sure this is needed? I don't think we should accept `yarn start` taking more than a minute, even in CI :p

File: packages/cli/src/commands/createPlugin.ts [hunk 2]
```diff
@@ -332,8 +332,16 @@ const createPlugin = async () => {
       type: 'input',
       name: 'id',
       message: chalk.blue('Enter an ID for the plugin [required]'),
-      validate: (value: any) =>
-        value ? true : chalk.red('Please enter an ID for the plugin'),
+      validate: (value: any) => {
+        if (!value) {
+          return chalk.red('Please enter an ID for the plugin');
+        } else if (!/^[-_a-zA-Z0-9]+$/.test(value)) {
```
Comment (Rugvip) [inline packages/cli/src/commands/createPlugin.ts]: I think we could go with `/^[a-z0-9]+(-[a-z0-9]+)$/i`? i.e. enforcing kebab-case and no leading, trailing or repeated `-`

File: scripts/cli-e2e-test.js (hunk 1 above)
Comment (freben) [reply scripts/cli-e2e-test.js]: Well it kept failing until i bumped it. Just the creation of the plugin is 30+ seconds alone.

File: packages/cli/src/commands/createPlugin.ts (hunk 2 above)
Comment (freben) [reply packages/cli/src/commands/createPlugin.ts]: but then the second parenthesis needs a `*` on it. sure i can do that

**Comment by freben** [comment]
@Rugvip updated

File: scripts/cli-e2e-test.js (hunk 1 above)
Comment (Rugvip) [reply scripts/cli-e2e-test.js]: Did it fail with the message "Failed to load page ..., max number of attempts reached"?

The timeout isn't started until the plugin creation is complete, so the timeout is just for `yarn start` to pick up the new plugin, build, and start serving it

Comment (freben) [reply scripts/cli-e2e-test.js]: https://github.com/spotify/backstage/runs/516185841?check_suite_focus=true

this is a failed run. it just bails, it seems. if it isn't this timeout, then it's something else - but so far, bumping this number has seemed to fix the problem consistently

**Comment by freben** [comment]
Went back with the timing ... dunno what was up but now it builds.

**Comment by Rugvip** [comment]
Keeping an :eye: on it, adding some more checks in https://github.com/spotify/backstage/pull/304 to make it break in better ways :grin:

**Pull request merged**